packages/discord-verify/src/lib/verify.ts

```diff
@@ -0,0 +1,167 @@
+import type {
+	Request,
+	SubtleCrypto,
+	SubtleCryptoImportKeyAlgorithm,
+} from "../types";
+
+export declare class TextEncoder {
+	constructor();
+	encode(input?: string): Uint8Array;
+}
+
+const encoder = new TextEncoder();
+
+const KEYS = {
+	ZERO: 48,
+	A: 65,
+	a: 97,
+};
+
+function hexCharToBinary(char: string) {
+	const code = char.charCodeAt(0);
+
+	if (code >= KEYS.a) {
+		return code - KEYS.a + 10;
+	}
+
+	if (code >= KEYS.A) {
+		return code - KEYS.A + 10;
+	}
+
+	return code - KEYS.ZERO;
+}
+
+/**
+ * Helper method that takes in a hex string and converts it to its binary representation.
+ * @param key Hex string to convert to binary
+ * @returns The binary form of a hex string
+ */
+export function hexStringToBinary(key: string | null) {
+	if (key == null) {
+		return new Uint8Array(0).buffer;
+	}
+
+	const view = new Uint8Array(key.length / 2);
+
+	for (let i = 0, o = 0; i < key.length; i += 2, ++o) {
+		view[o] = (hexCharToBinary(key[i]) << 4) | hexCharToBinary(key[i + 1]);
+	}
+
+	return view.buffer;
+}
+
+async function getCryptoKey(
```

Comment:
You docced the other thing, why not this?

Reply:
Internal vs External function - only added documentation to external functions